app/main.py

```diff
@@ -1,2 +1,168 @@
+        return self.__current_len == load_factor
+
+    def items(self) -> list:
+        return [item for item in self.__items if item is not None]
+
+    def __setitem(self, key: Any, value: Any) -> None:
+        current_index = self.__get_index_from_hash(key)
+
+        while True:
+            if not self.__items[current_index]:
+                self.__items[current_index] = (key, value)
+                self.__current_len += 1
+                return
+            elif self.__items[current_index][0] == key:
+                self.__items[current_index] = (key, value)
+                return
+            else:
+                current_index += 1
+                if current_index == len(self.__items):
+                    current_index = 0
+
+    def __repr__(self) -> str:
+        dict_str_list: List[str] = []
+        for key, value in self.items():
+            if isinstance(value, str):
+                dict_str_list.append(f"{key}: '{value}'")
+            else:
+                dict_str_list.append(f"{key}: {value}")
+
+        dict_str = ", ".join(dict_str_list)
+        return f"{{{dict_str}}}"
+
+    def clear(self) -> None:
+        self.__items = []
+        self.__current_len = 0
+
+    def __delitem__(self, key: Any) -> None:
+        current_index = self.__find_item_index_by_key(key)
+        self.__del_item_by_index(current_index)
+
+    def __del_item_by_index(self, current_index: int) -> None:
+        self.__current_len -= 1
+        self.__items[current_index] = None
+
+    def __get_index_from_hash(self, key: Any) -> int:
+        key_hash = hash(key)
+        length = len(self.__items)
+        return key_hash % length
+
+    def __find_item_index_by_key(self, key: any) -> int:
+        current_index = self.__get_index_from_hash(key)
+        begin_index = None
+        while True:
+            if (
+                    self.__items[current_index] is not None
+                    and self.__items[current_index][0] == key
+            ):
+                return current_index
+            else:
+                if begin_index is None:
+                    begin_index = current_index
+                elif begin_index == current_index:
+                    raise KeyError(
+                        f"Dictionary with key '{key}' is not exists"
+                    )
+                current_index += 1
+                if current_index == len(self.__items):
+                    current_index = 0
+
+    def get(self, key: Any) -> Any:
+        return self.__getitem__(key)
+
+    def pop(self, key: Any) -> Any:
+        current_index = self.__find_item_index_by_key(key)
+        item_value = self.__items[current_index][1]
+        self.__del_item_by_index(current_index)
+        return item_value
+
+    def update(self, other_dict: Any) -> None:
+        if isinstance(other_dict, Dictionary):
+            other_dict_items = other_dict.items()
+        else:
+            other_dict_items = other_dict
```

Comment:
There is no need in this `else` statement, just rewrite the content of `other_dict` in the first if condition.